### DISCUSSION TITLE
Latin square

### DISCUSSION THREAD

**Discussion by g-porta** [jspsych/jsPsych#3252] [Q&A]
Hi! 
My experiment uses a Latin square design. I created each list, and now I want the code to assign each list to a participant. I tried fully randomising this part, but the results are too random, and I'd be force to delate some data. 

Is there a way of alternating between four lists automatically? 

Thanks!

**Comment by jodeleeuw** [comment]
Hi @g-porta,

How are you hosting your experiment online? (This is a problem that has to be solved on the hosting side.)

**Reply by g-porta** [reply]
Hi! 

I'm hosting in it on Jatos. 

I tried to split the lists into four components using `jatos.startNextComponent `, but I couldn't get it to work. I tried modifying the template they put online ([Randomize Tasks Between Workers](https://github.com/JATOS/JATOS_examples/raw/main/examples/randomize_tasks_between_workers.jzip)) but I couldn't get that one to work either. 

I tried with the batch function, but I still get some kind of error. 

```ruby
function initBatchConditions() {
      if (!jatos.batchSession.defined("/trials")) {
        jatos.batchSession.set("trials", 0).fail(initBatchConditions);
      }
      
      trials = jatos.batchSession.get("trials")
      jatos.batchSession.set("trials", trials + 1 ).fail(initBatchConditions); // If it fails: try again
        console.log(jatos.batchSession.getAll());
        console.log(trials % groups.length);
      return trials;
    }

```

I don't really know where to go from here...  

What would you recommend?

Thanks!

**Reply by alessandraluz** [reply]
Hi @g-porta, did you figure out a fix? I am stuck on this too.
